packages/discovery/src/discovery/output/saveDiscoveryResult.ts

```diff
@@ -27,6 +28,51 @@ export interface SaveDiscoveryResultOptions {
   flatSourcesFolder?: string
   discoveryFilename?: string
   metaFilename?: string
+  skipHints?: boolean
+}
+
+export async function saveConfigHints(
+  rootPath: string,
+  results: Analysis[],
+  options: SaveDiscoveryResultOptions,
+): Promise<void> {
+  const filePath = path.join(rootPath, 'config.hints.jsonc')
+  await rimraf(filePath)
+  if (options.skipHints) {
+    return
+  }
+
+  const contractsWithMatchedTemplates = results.filter(
+    (r) => r.type === 'Contract' && !isEmpty(r.matchingTemplates),
+  ) as AnalyzedContract[]
+
+  if (contractsWithMatchedTemplates.length === 0) {
+    return
+  }
+
+  const content = [
```

Comment:
Maybe it's worth to do it nicer? Either use something like [comment-json](https://www.npmjs.com/package/comment-json) or just `JSON.stringify()` the object and append comments at the top.

Reply:
I like the idea of using comment-json here but the implementation I'm getting is even more complex and it's difficult to make it work nicely with TypeScript.

I've agreed with @mateuszradomski that we will keep current implementation as is and refactor it if we decide to keep using config.hint.jsonc file in the furure (see: L2B-5274)